deploy/common/k8sbased.sh

```diff
@@ -672,13 +672,29 @@ function launch_sensor {
         exit 1
       fi
       mkdir "$k8s_dir/sensor-deploy"
-      touch "$k8s_dir/sensor-deploy/init-bundle.yaml"
-      chmod 0600 "$k8s_dir/sensor-deploy/init-bundle.yaml"
-      curl_central "https://${API_ENDPOINT}/v1/cluster-init/init-bundles" \
-          -XPOST -d '{"name":"deploy-'"${CLUSTER}-$(date '+%Y%m%d%H%M%S')"'"}' \
-          | jq '.helmValuesBundle' -r | base64 --decode >"$k8s_dir/sensor-deploy/init-bundle.yaml"
+      (umask 077; touch "$k8s_dir/sensor-deploy/init-bundle.yaml")
+      local api_resp
+      api_resp="$(mktemp)"
+      # Retry with different bundle name to recover from situations where bundle was created on
+      # central but curl failed to read response for whatever reason.
+      for _ in $(seq 10); do
+        if curl_central_once "https://${API_ENDPOINT}/v1/cluster-init/init-bundles" \
+            -o "${api_resp}" \
+            -XPOST -d '{"name":"deploy-'"${CLUSTER}-$(date '+%Y%m%d%H%M%S')"'"}'; then
+          break
+        else
+          : > "${api_resp}"
```

Comment:
Could be useful for debugging to echo here

Reply:
The echo is on like 692. Making this file empty is just a way to propagate the error.